### DISCUSSION TITLE
Create new file from template

### DISCUSSION THREAD

**Discussion by Ampa** [SilentVoid13/Templater#514] [Ideas & Feature Requests]
At present we can create a new file, and then as a second step insert a template.

Would you consider a new action: Create new from template?

This would carry out both steps as a single action.

**Comment by SilentVoid13** [comment]
Hey @Ampa, this has been implemented in 1.1.0. Update and you should have a new hotkey that allows you to create a new note from template. 
Tell me if it works for you or if you have any suggestions before I close this.

**Comment by luckman212** [comment]
@SilentVoid13 this is a great addition, thank you! 

I hit a bug though, the default hotkey <kbd>⌥opt N</kbd> doesn't work on macOS - it only produces a `˜` tilde character in the editor. I changed it to <kbd>⌘cmd ⇧shift N</kbd> and that seems to work.

**Comment by Ampa** [comment]
@SilentVoid13 Thanks for adding the new feature.

For some reason I am unable to get it to work. Win10, Obsidian 0.11.13

The modal dialog appears, I select a template, and either click with mouse, or hit enter.
Neither creates the new file.

😟

**Comment by rishifter** [comment]
This sounds like a wonderful feature but I think there's an issue with how it works if you use the tp.file.title in any of your templates. It always gets replaced to Untitled which is the default filename instead of what we name a note. However the tp.file.cursor still works. Perhaps there could be a way to bind the same shortcut to defer replacing certain template items like title in this case?

**Comment by Ampa** [comment]
> I am unable to get it to work.

Found the problem...

**New from Template will only work if there is not a file called untitled.md**

**Comment by SilentVoid13** [comment]
Hey @Ampa, thanks for the feedback!

This should be fixed in Templater 1.3.0. Templater will also now create new notes in the folder defined in obsidian settings.

Can you update and tell me if this works now ?

**Comment by Ampa** [comment]
Updated, and... looks like you fixed it!

Thank you.

**Comment by Moonbase59** [comment]
It would be great if something like "New note from template" could be added to Obsidian file explorer’s right-click menu, too.

Reason: I often create new notes in subfolders and use the right-click menu on the subfolder to add a new note there.

Using Obsidian 0.11.13 on Linux Mint/Cinnamon 20.1, Templater version 1.3.0.

**Comment by SilentVoid13** [comment]
Hey @Moonbase59, this has been implemented in Templater 1.4.0!
Update and tell me if this works for you.

**Comment by nnkn** [comment]
Hi @SilentVoid13 ,There is a little bug, when I create a new note from template，it will apper the modal dialog, even if there is only one template. 
Can I just create a new note without the modal dialog when there is only one template? or a hotkey to create a new note with specified template.

**Comment by Moonbase59** [comment]
> Hey @Moonbase59, this has been implemented in Templater 1.4.0!
> Update and tell me if this works for you.

Works like a charm, thanks for implementing this!

**Comment by tobulli** [comment]
> This sounds like a wonderful feature but I think there's an issue with how it works if you use the tp.file.title in any of your templates. It always gets replaced to Untitled which is the default filename instead of what we name a note. However the tp.file.cursor still works. Perhaps there could be a way to bind the same shortcut to defer replacing certain template items like title in this case?

Is this solved yet? Maybe having a prompt to give it a title before the whole template is inserted would be great.

**Comment by frankreporting** [comment]
Successfully creating new notes from template. However, the new notes keep appearing in my vault root, rather than the same folder as my current file (which is what I have set in my Obsidian settings).

**Comment by nnkn** [comment]
> Hi @SilentVoid13 ,There is a little bug, when I create a new note from template，it will apper the modal dialog, even if there is only one template.
> Can I just create a new note without the modal dialog when there is only one template? or a hotkey to create a new note with specified template.

hi，@SilentVoid13 after I update to the latest 1.6.0 ,the bug appears again...

**Comment by joellti** [comment]
I noticed the same behavior with 1.6.0. But if I invoke the new note action from the right-click menu it was created in the folder as expected.

> Successfully creating new notes from template. However, the new notes keep appearing in my vault root, rather than the same folder as my current file (which is what I have set in my Obsidian settings).

**Comment by joellti** [comment]
> > This sounds like a wonderful feature but I think there's an issue with how it works if you use the tp.file.title in any of your templates. It always gets replaced to Untitled which is the default filename instead of what we name a note. However the tp.file.cursor still works. Perhaps there could be a way to bind the same shortcut to defer replacing certain template items like title in this case?
> 
> Is this solved yet? Maybe having a prompt to give it a title before the whole template is inserted would be great.

I would like to see that as well. Either prompt for a title or even better, fill in the title automatically using a tp command

**Comment by itpropro** [comment]
> > > This sounds like a wonderful feature but I think there's an issue with how it works if you use the tp.file.title in any of your templates. It always gets replaced to Untitled which is the default filename instead of what we name a note. However the tp.file.cursor still works. Perhaps there could be a way to bind the same shortcut to defer replacing certain template items like title in this case?
> > 
> > 
> > Is this solved yet? Maybe having a prompt to give it a title before the whole template is inserted would be great.
> 
> I would like to see that as well. Either prompt for a title or even better, fill in the title automatically using a tp command

Hey @SilentVoid13, do you have any tips how to handle filenames (file.title) in this case? I would like to create a new note from templater and directly generate the filename from the current date plus and input prompt.

EDIT: Solved it with JS, took 1 minute, great work with the plugin 👍

```Javascript
<%*
    const folder = await tp.system.prompt("Folder")
    const title = await tp.system.prompt("Title")
    await tp.file.rename(`${title}`)
    await tp.file.move(`/${folder}/${title}`)
%>---
title: <%* tR += title %>
creation-date: <% tp.file.creation_date() %>
---
```

**Comment by nathonius** [comment]
> Hi @SilentVoid13 ,There is a little bug, when I create a new note from template，it will apper the modal dialog, even if there is only one template. Can I just create a new note without the modal dialog when there is only one template? or a hotkey to create a new note with specified template.

I would love to have a separate hotkey/command added to create a new file from a specific template, the same way you can do with inserting a specific template.

**Comment by itpropro** [comment]
> > Hi @SilentVoid13 ,There is a little bug, when I create a new note from template，it will apper the modal dialog, even if there is only one template. Can I just create a new note without the modal dialog when there is only one template? or a hotkey to create a new note with specified template.
> 
> I would love to have a separate hotkey/command added to create a new file from a specific template, the same way you can do with inserting a specific template.

You can already do that by using the built-in template hotkeys:
![image](https://user-images.githubusercontent.com/15030951/142767877-edd7d3d4-d955-4e6f-a6e0-9535c7925ce8.png)

**Comment by nathonius** [comment]
That only seems to add a command to insert the template into an existing file, not creating a new file from the template.

**Comment by itpropro** [comment]
> That only seems to add a command to insert the template into an existing file, not creating a new file from the template.

It totally works with my code from above. It asks you for the file name or whatever you define and creates the new file. I use that daily to create daily notes, meeting notes and all kinds of template based workflows. If you want to have a static name or a name based on the current date, etc. you can just edit the following part to match your workflow:
```
...
    await tp.file.rename(`${title}`)
    await tp.file.move(`/${folder}/${title}`)
...
```

**Reply by itssource** [reply]
This works for me! Thanks @itpropro. At first, I had forgotten to remove the initial shortcut combo set for my original attempt (with had the ...: insert code line in the Settings for Templater. 

A bonus feature for this example would be if we hit the 'x' button to close any of the dialogs, the file is not created. Because we may change our mind. So, stops processing input fields, then loading the file not needed.

Still, this is appreciated.

Update, careful!
I also learned. When you see 'Templater: Insert' in the Settings area for the "Configure Hotkey" panel, change that to Templater: ' You will effectively be deleting the 'Insert' word. Now you will see the default ALT+N shortcut. Which will allow you to show this new template (not Insert into existing document) template sample (and to change the shortcut to something on the left side of the keyboard if you are also right-handed, for ease of use?).

This one tripped me up for a few minutes.
